### ISSUE TITLE
Enforcing props destructuring

### ISSUE THREAD

**Issue by Ericlm** [vuejs/eslint-plugin-vue#2688]
**Please describe what the rule should do:**
The role of this rule would be to suggest [props destructuring](https://vuejs.org/guide/components/props.html#reactive-props-destructure). It could also warns on its usage if necessary.

**What category should the rule belong to?**

[X] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
// vue/prefer-props-destructuring

const props = defineProps<>({...}) // 🚨

const {...} = defineProps<>({...}) // 👍
```

**Additional context**

**Comment by mrleblanc101** [comment]
Since reactive props destructuring has been added in Vue 3.5, I think a rule for this is important for code consistency !

**Comment by waynzh** [comment]
This is certainly a useful rule. It's pretty straightforward, given that a function already exists to identify usage of prop destructuring.
The rule is named `enforce-prop-destructuring`. Perhaps we can disregard the options - if someone wants to utilize this rule, they can simply apply it.

**Comment by ZexTech** [comment]
There is a false positive with this rule, flagging errors when the props variable or destructured properties are not defined.

Example
```
// Error ✅
const props = withDefaults(
  defineProps<{
    a?: boolean;
  }>(),
  {
    a: false,
  }
);

// Valid ✅
const { a = false } = defineProps<{ a?: boolean }>();

// False positive ❌
defineProps<{ a?: boolean }>();
```

**Comment by esfahanweb** [comment]
Currently, the rule only supports two options: `"always"` and `"never"`.  
But in many Composition API use-cases, `defineProps()` is used *only* for template usage — without assigning it to a variable. In those cases, destructuring is irrelevant and should not be enforced.

This leads to unnecessary ESLint errors when the rule is set to `"always"`.


Add a new option, for example:

```js
"vue/define-props-destructuring": [
  "error",
  {
    destructure: "only-when-assigned"
  }
]

**Comment by ota-meshi** [comment]
Please do not comment on your problem on a closed issue.
Open a new issue and provide details about your problem and a reproduction repository.
